### PR TITLE
fix: use absoultePath as fileCache key

### PR DESCRIPTION
**Pull request by TrickyPi**
## Motivation
exist different key of the same file
like this:
image

### CODE REVIEW

**Comment by chu121su12** [comment]
For cli test & snapshot, please run `pnpm bulid` first

**Comment by TrickyPi** [comment]
Thanks for your reply, i had run this command before committing.

**Review by antfu — APPROVED** [review]

File: test/__snapshots__/cli.test.ts.snap [hunk 1]
```diff
@@ -19,5 +19,5 @@ exports[`cli > uno.css exclude initialized class after changing file 1`] = `
...
 /* layer: default */
-.bg-red{--un-bg-opacity:1;background-color:rgba(248,113,113,var(--un-bg-opacity));}"
+.bg-blue{--un-bg-opacity:1;background-color:rgba(96,165,250,var(--un-bg-opacity));}"
```
Comment (TrickyPi) [inline test/__snapshots__/cli.test.ts.snap:22]: @antfu the expected className is `.bg-red`,see here https://github.com/unocss/unocss/pull/1045/files#diff-1deee0a575599b2df117c280da319f7938aaf6fdb0c04bcdbde769dbf464be69R64

Comment (TrickyPi) [reply test/__snapshots__/cli.test.ts.snap:22]: Hmmm! that make sense

Comment (TrickyPi) [reply test/__snapshots__/cli.test.ts.snap:22]: Should I create a new PR to change this test?

**Pull request merged**